GH-3092 Refactor FTP and SFTP outbound specs, handlers

deepak5127 opened the pull request:
Following are the changes,

1. Deprecated SFTP and FTP classes outboundAdapter method which takes RemoteFileTemplate. Instead created new methods in both classes which takes SftpRemoteFileTemplate and FtpRemoteFileTemplate respectively.
2. Deprecated SftpMessageHandlerSpec's constructor which takes RemoteFileTemplate and added new constructors which takes SftpRemoteFileTemplate
3. Deprecated FtpMessageHandlerSpec's constructor which takes RemoteFileTemplate and added new constructors which takes FtpRemoteFileTemplate. Deprecated FtpMessageHandler's constructor which takes RemoteFileTemplate and added new constructor with ftpRemoteFileTemplate.
4. Added new test cases in FtpTests and SftpTests


Fixes https://github.com/spring-projects/spring-integration/issues/3092

deepak5127 committed:
GH-3092 Refactor FTP and SFTP outbound specs, handlers

artembilan commented on spring-integration-ftp/src/main/java/org/springframework/integration/ftp/dsl/Ftp.java:
```diff
@@ -121,13 +123,36 @@ public static FtpMessageHandlerSpec outboundAdapter(RemoteFileTemplate<FTPFile>
+	/**
+	 * A {@link FtpMessageHandlerSpec} factory for an outbound channel adapter spec.
+	 * @param ftpRemoteFileTemplate the remote file template.
+	 * @return the spec.
```
Please, add `@since 5.4` for all new public API

deepak5127 replied:
Added 5.4 version to new methods

artembilan commented on spring-integration-ftp/src/main/java/org/springframework/integration/ftp/outbound/FtpMessageHandler.java:
```diff
@@ -48,10 +48,15 @@ public FtpMessageHandler(FtpRemoteFileTemplate remoteFileTemplate) {
 		super(remoteFileTemplate);
 	}
 
+	@Deprecated
```
Please, add some JavaDocs here with the `@deprecated` tag and explanation `in favor of ...`

deepak5127 replied:
Added

artembilan replied:
No reason to spam me with `Added` for all my comments.
If you agree with the change requested, you just push the commit.
We get notifications for them.
You comment only if you are not agree with us 😄

artembilan commented on spring-integration-ftp/src/main/java/org/springframework/integration/ftp/outbound/FtpMessageHandler.java:
```diff
@@ -48,10 +48,15 @@ public FtpMessageHandler(FtpRemoteFileTemplate remoteFileTemplate) {
 		super(remoteFileTemplate, mode);
 	}
 
+	public FtpMessageHandler(FtpRemoteFileTemplate ftpRemoteFileTemplate, FileExistsMode mode) {
```
Please, add some JavaDocs to all the new public API.

deepak5127 replied:
Added

artembilan commented on spring-integration-sftp/src/main/java/org/springframework/integration/sftp/dsl/Sftp.java:
```diff
@@ -122,13 +124,36 @@ public static SftpMessageHandlerSpec outboundAdapter(RemoteFileTemplate<ChannelS
+	/**
+	 * An {@link SftpMessageHandlerSpec} factory for an outbound channel adapter spec.
+	 * @param sftpRemoteFileTemplate the remote file template.
+     * @return the spec.
```
Please, format JavaDocs properly.
And of course `@since 5.4`!

deepak5127 replied:
Added

artembilan reviewed (changes requested):
Please, add your name to the `@author` list of all the affected classes.

Would you also mind to take a look into the `(s)ftp.adoc` to be sure that code snippets over there use already an new API.

Great job so far, thank you!

deepak5127 committed:
GH-3092 Incorporate review comments

deepak5127 committed:
GH-3092 Incorporate review comments

deepak5127 commented:
@artembilan  Added @author for all the files I have changed. (s)ftp.adoc does not reference any code related to RemoteFileTemplate. Please check all the comments I have incorporated and let me know if I miss something

artembilan commented on spring-integration-ftp/src/main/java/org/springframework/integration/ftp/dsl/FtpMessageHandlerSpec.java:
```diff
@@ -30,7 +30,7 @@
  * @author Joaquin Santana
- *
+ * @author Deepak Gunasekaran
  * @since 5.0
```
There must be a blank line before `@since` on the class level JavaDocs. This way the author list is much visible.

deepak5127 committed:
GH-3092 Incorporate review comments

artembilan commented:
@deepak5127 ,

thank you for contribution; looking forward for more!

deepak5127 commented:
Thanks @artembilan  for the guidance. Please also let me know if you have anything you feel I can work on, meanwhile I will check on other open issues.

artembilan commented:
Right. See here:

https://github.com/spring-projects/spring-integration/issues/3194
https://github.com/spring-projects/spring-integration/issues/3306

Also there are still a lot of tests which should got to JUnit 5 from their JUnit 4.
